src/chapters/loops/iteration.rst

```diff
@@ -0,0 +1,43 @@
+Iteration
+=========
+
+.. index:: ! iteration
+
+When repeating the same action over and over again, a human is likely to make a mistake at some point. Computers, however, possess the incredible ability to carry out repetitive tasks without making mistakes. 
+
+To see this, let's consider an appropriate, if somewhat contrived, example. Suppose you want to print out the integers 0 through 50. With the tools you currently have at your disposal, your program would look like this:
+
+.. sourcecode:: js
+
+   console.log(0);
+   console.log(1);
+   console.log(2);
+   console.log(3);
+   console.log(4);
+   // and so on...
+
+Not only is this highly repetetive, but it is also error-prone. Even if utilizing copy-paste functionality, the sheer volume of code makes it somewhat likely that we will make a simple mistake, such as skipping an integer or misspelling ``console``.
+
+This code is also hard to modify. If we want to make a conceptually simple change---such as printing all the way to 100, or only printing even numbers---then we are forced to update an immense amount of code. Programming languages provide tools that allow us to repeat a sequence of statements in a much simpler way that this.
+
+.. index::
+   single: loop; for
+   single: loop; while
+
+Repeated execution of a sequence of statements is called **iteration**. This chapter explores two mechanisms that JavaScript provides to make iteration simple and flexible---the ``for`` and ``while`` loops.
+
+To give you a taste of what's to come, here is how we could write the program above using a ``for`` loop.
+
+.. sourcecode:: js
+
+   for (let i = 0; i < 51; i++) {
+      console.log(i);
+   }
+
```

Comment:
Many times, I've seen examples where the ‘let’ in the 1st statement in the () is omitted, and JS infers its usage.  How strict do we want to be here?  Dropping the ‘let’ cleans up the code appearance for novices, but does it conflict with our style guidelines?

Reply:
Correct, if no keyword is used then `let` is inferred. We haven't discussed a convention/style preference. I lean toward being explicit (obviously) since I think it makes it more clear to students that a variable is being declared there, and nothing fishy is happening behind the scenes. But I think this would be good to discuss. I'll slot it into our next meeting.